libfwupdplugin/fu-device-event.c

```diff
@@ -73,11 +79,52 @@ fu_device_event_blob_new(GType gtype, const gchar *key, gpointer data, GDestroyN
 	return blob;
 }
 
+static gboolean
+fu_device_event_id_validate(const gchar *id)
+{
+	if (strlen(id) != FU_DEVICE_EVENT_KEY_HASH_PREFIX_SIZE)
+		return FALSE;
+	for (guint i = 0; id[i] != '\0'; i++) {
+		if ((id[i] < 'a' || id[i] > 'f') && (id[i] < '0' || id[i] > '9'))
+			return FALSE;
+	}
```

Comment:
to avoid this whole dance, what would you think about making `FU_DEVICE_EVENT_KEY_HASH_PREFIX_SIZE` 9 characters and putting a character at the start or end that is unique to fwupd's use?  Something like a `!`.  Then you can look for that sentinel and the length and call it good enough.

Reply:
I think a prefix (`#`, for hash!) is a good idea; and we can leave the length unchecked in case we do want to make it longer in the future.

Reply:
Fixed.